File: server/src/main/java/com/khrystyna/nerdysoft/models/Task.java
package com.khrystyna.nerdysoft.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.time.LocalDateTime;
import java.util.LinkedList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    private String id;
    private String title;
    private String description;
    @LastModifiedDate
    private LocalDateTime dateTime;
    @DBRef
    private User author;
    @DBRef
    private List<User> users = new LinkedList<>();
}

File: client/src/app/models/entity/task.ts
import {User} from './user';

export class Task {
  constructor(
    public id?: string,
    public title?: string,
    public description?: string,
    public dateTime?: Date,
    public author?: User,
    public users?: User[],
    public share = false,
    public edit = false
  ) {
  }
}
